mlrun/feature_store/api.py

```diff
@@ -780,16 +780,14 @@ def _ingest_with_spark(
 
         key_columns = list(featureset.spec.entities.keys())
         timestamp_key = featureset.spec.timestamp_key
-        if not targets:
-            if not featureset.spec.targets:
-                featureset.set_targets()
-            targets = featureset.spec.targets
-            targets = [get_target_driver(target, featureset) for target in targets]
+        targets = targets or featureset.spec.targets
```

Comment:
Note that by dropping the `featureset.set_target()` call, the targets are not stored in the fset spec anymore. I think we need to call this function, probably in the `ingest` function itself, WDYT?

Reply:
Hm, this case was only relevant when targets weren't defined at all (default targets). So I don't think we need to keep them in the spec in this case.